plugin/plugin_test.go

```diff
@@ -0,0 +1,130 @@
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockParser struct {
+	flags                 []*flag
+	shouldErr             bool
+	errOnCallCount        int
+	addFlagGroupCallCount int
+}
+
+func (p *mockParser) AddFlagGroup(short, long string, data interface{}) error {
+	if p.shouldErr && p.errOnCallCount == p.addFlagGroupCallCount {
+		return errors.New("bad")
+	}
+	p.flags = append(p.flags, &flag{
+		shortDescription: short,
+		longDescription:  long,
+		data:             data,
+	})
+	p.addFlagGroupCallCount++
+	return nil
+}
+
+type fooFlags struct {
+	Bar string `long:"bar" description:"Sets the 'bar' value."`
+}
+
+// defer-able test cleanup for global state
+func resetFlags(_flags []*flag) func() {
+	flags = _flags
```

Comment:
Isn't this setFlags with a `reset` return?

also, nit: local variables shouldn't be `_` prefixed